demo/config/environments/test.rb

```diff
@@ -14,10 +14,10 @@
 
   config.cache_classes = false
 
-  # Do not eager load code on boot. This avoids loading your whole application
-  # just for the purpose of running a single test. If you are using a tool that
-  # preloads Rails for running tests, you may have to set it to true.
-  config.eager_load = false
+  # Eager loading loads your whole application. When running a single test locally,
+  # this probably isn't necessary. It's a good idea to do in a continuous integration
+  # system, or in some way before deploying your code.
+  config.eager_load = ENV["CI"].present?
```

Comment:
This is pinched from new Rails 7 app as the description and the behaviour is exactly what we want. Useful to test with eager loading enabled as that's more prod-like but only when running in CI environment.